NewKernel_d/include/CGAL/NewKernel_d/KernelD_converter.h

```diff
@@ -114,19 +115,25 @@ class KernelD_converter_<Final_,K1,K2,typeset<> > {
 
 
 // TODO: use the intersection of Kn::Object_list.
-template<class K1, class K2, class List_=
-typename typeset_intersection<typename K1::Object_list, typename K2::Object_list>::type
+template<class K1, class K2,
+         class List_= Default,
+         typename NTc = NT_converter<typename Get_type<K1, FT_tag>::type, typename Get_type<K2, FT_tag>::type>
```

Comment:
IIUC:
* You want to be able to convert from an approximate (Interval_nt) kernel to an exact kernel: can you explain in what conditions this is used? Are the intervals guaranteed to be points, or is any value in the interval ok for your purpose?
* NT_converter does not support conversion from Interval_nt to anything but double: if converting to something else makes sense, shouldn't we specialize NT_converter?
* Instead of passing a functor that converts to the rational type, you pass one that converts to double and rely on an implicit conversion happening afterwards. At first this feels like a hack, but problems seem unlikely (and it might even have slightly better performance than the "clean" solution for some kernels), so that's ok.

Reply:
Ok, if the intervals are points, the conversion makes sense. To go back to my second point: what do you think of specializing `NT_converter<Interval_nt,X>` so that
1. it asserts that the interval is a point (CGAL_precondition?)
2. it chains `NT_converter<Interval_nt,double>` (or actually calls `.sup()` directly, no need to average 2 identical values) and `NT_converter<double,X>`

?
Adding a parameter to the kernel converter is not a bad idea, but I think it should only happen if we first reject the simpler solution (maybe you already did and I simply missed earlier discussions).

Reply:
Personally  I find the NT converter as template parameter cleaner than the specialization.  Also I like that in dD it is like in 2D/3D.

Reply:
* Is there any real use of that third parameter in 2D/3D? The only uses I found (not an exhaustive search) were with To_double or To_interval and they look like they would probably work just as well with the default parameter.
* Having a parameter does make it easier to have different versions, for instance this one that expects a trivial interval vs what I did in (the abandoned) Epack_d where I accepted conversions depending on `has_smaller_relative_precision`. So :+1: for the extra parameter.
* I guess this just leaves the comment in https://github.com/CGAL/cgal/pull/8284#discussion_r1765255498 about replacing `NT_converter<distance_t,double>` with something more specific at some point in the future.